### DISCUSSION TITLE
COCO8-seg - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#5377] [Docs]
# COCO8-seg - Ultralytics YOLOv8 Docs

Discover the COCO8-Seg: a compact but versatile instance segmentation dataset ideal for testing Ultralytics YOLOv8 detection approaches. Complete usage guide included.

https://docs.ultralytics.com/datasets/segment/coco8-seg/

**Comment by digogogo** [comment]
what should i do to slove this question

Traceback (most recent call last):
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 418, in _smart_load
    return self.task_map[self.task][key]
  File "D:\python\miniconda\envs\yolov8\lib\site-packages\torch\nn\modules\module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'YOLO' object has no attribute 'task_map'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\yolov8\ultralytics-main\coco-train.py", line 7, in <module>
    results = model.train(data='coco8.yaml', epochs=100, imgsz=640)
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 336, in train
    self.trainer = (trainer or self._smart_load('trainer'))(overrides=args, _callbacks=self.callbacks)
  File "D:\yolov8\ultralytics-main\ultralytics\engine\model.py", line 422, in _smart_load
    raise NotImplementedError(
NotImplementedError: WARNING  'YOLO' model does not support 'train' mode for 'detect' task yet.

**Reply by pderrenger** [reply]
@digogogo it appears that you're encountering an issue where the 'YOLO' object does not support the 'train' mode for the 'detect' task. This could be due to a mismatch between the task specified in your model configuration and the operation you're trying to perform.

To resolve this issue, please ensure that your model configuration aligns with the task you want to perform. If you're attempting to train a model for object detection, make sure that the task is set correctly in your model's YAML configuration file.

If you continue to face difficulties, consider reviewing the documentation for the correct usage of training mode and ensure that your environment is set up correctly with all necessary dependencies installed. If the problem persists, you might want to check for updates or report the issue on the repository for further assistance.
